### PR TITLE
#1090 - Better support for L2

### PR DESCRIPTION
**Pull request by ddesrosier**
- Block version checks for L2 for now
- Add custom yield source warning
- Add connect to other networks button
- Better WETH logo
- Surface "start award" and "complete award" in prize card so anyone can call it easily

Screen Shot 2021-03-24 at 2 35 24 PM
Screen Shot 2021-03-24 at 2 35 48 PM

### CODE REVIEW

**Comment by ddesrosier** [comment]
Blocked on https://github.com/pooltogether/etherplex/pull/11

File: lib/components/BlockExplorerLink.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import FeatherIcon from 'feather-icons-react'
+import classnames from 'classnames'
+import React from 'react'
+
+import { formatBlockExplorerAddressUrl, formatBlockExplorerTxUrl } from 'lib/utils/networks'
+import { shorten as shortenHash } from 'lib/utils/shorten'
+import { useNetwork } from 'lib/hooks/useNetwork'
+
+export const BlockExplorerLink = (props) => {
```
Comment (chuckbergeron) [inline lib/components/BlockExplorerLink.jsx:9]: 💥

File: lib/components/IndexContent.jsx [hunk 2]
```diff
@@ -79,14 +81,32 @@ const PoolsLists = () => {
...
+
+  if (![NETWORK.matic, NETWORK.mumbai].includes(chainId)) return null
+
+  const networkName = NETWORK_DATA[chainId].view
+
+  return (
+    <div className='text-left mb-10 border-2 border-primary rounded-lg px-7 py-4 text-accent-1'>
+      <CardTitle noMargin>☹️ Index Unavailable for {networkName}</CardTitle>
+      <p>
+        Unfortunately due to limitations of MaticVigil we can't dynamically compile a list of
```
Comment (chuckbergeron) [inline lib/components/IndexContent.jsx:103]: Let's just call this "Matic" instead of "MaticVigil"

File: lib/hooks/useAddNetworkToMetamask.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { NETWORK_DATA } from 'lib/utils/networks'
+import { useContext, useMemo } from 'react'
+
+export const useAddNetworkToMetamask = (chainId) => {
+  const walletContext = useContext(WalletContext)
+
+  // TODO: hide for rinkeby, kovan etc
+  console.log(walletContext)
+
+  return useMemo(() => {
```
Comment (chuckbergeron) [inline lib/hooks/useAddNetworkToMetamask.js:11]: Do you think this needs useMemo or would benefit of it? I'm wary of using it too much when it's unnecessary unless it actually shows a reasonable performance improvement.

File: lib/hooks/useDetermineContractVersions.jsx [hunk 4]
```diff
@@ -129,6 +134,8 @@ const IncompatibleContract = (props) => {
 
   if (hideWarning) return null
 
+  if (NETWORKS_WITHOUT_VERSIONING.includes(chainId)) return null
```
Comment (chuckbergeron) [inline lib/hooks/useDetermineContractVersions.jsx:137]: What do you mean by "VERSIONING" here?

**Review by chuckbergeron — CHANGES_REQUESTED** [review]

**Comment by chuckbergeron** [comment]
Changed this so it fit a bit better with the rest of the theme and isn't so daunting. Should still grab attention tho, hopefully. :)
Screenshot 2021-03-25 at 13 42 19

File: lib/hooks/useAddNetworkToMetamask.js (hunk 3 above)
Comment (ddesrosier) [reply lib/hooks/useAddNetworkToMetamask.js:11]: https://blog.logrocket.com/rethinking-hooks-memoization/ yeah might be over using it

**Review by chuckbergeron — APPROVED** [review]

**Pull request merged**